drizzle-orm/src/mysql-core/query-builders/set-operators.ts

```diff
@@ -0,0 +1,351 @@
+> = TValue extends MySqlSetOperatorBuilder<any, infer TSel, infer TMode, any, infer TNull> ? ValidateShape<
+		SelectResult<TSel, TMode, TNull>,
+		SelectResult<TSelection, TSelectMode, TNullabilityMap>,
+		TypedQueryBuilder<any, SelectResult<TSelection, TSelectMode, TNullabilityMap>[]>
+	>
+	: TValue;
+
+type SetOperatorRestSelect<
+	TValue extends readonly TypedQueryBuilder<any, any[]>[],
+	Valid,
+> = TValue extends [infer First, ...infer Rest]
+	? First extends MySqlSetOperatorBuilder<any, infer TSel, infer TMode, any, infer TNull>
+		? Rest extends TypedQueryBuilder<any, any[]>[] ? [
+				ValidateShape<SelectResult<TSel, TMode, TNull>, Valid, TValue[0]>,
+				...SetOperatorRestSelect<Rest, Valid>,
+			]
+		: ValidateShape<SelectResult<TSel, TMode, TNull>, Valid, TValue>
+	: never[]
+	: TValue;
+
+export interface MySqlSetOperatorConfig {
+	fields: Record<string, unknown>;
+	operator: SetOperator;
+	isAll: boolean;
+	leftSelect: MySqlSetOperatorBuilder<any, any, any, any, any>;
+	rightSelect: TypedQueryBuilder<any, any[]>;
+	limit?: number | Placeholder;
+	orderBy?: (MySqlColumn | SQL | SQL.Aliased)[];
+	offset?: number | Placeholder;
+}
+
+export interface MySqlSetOperatorBuilder<
+	TTableName extends string | undefined,
+	TSelection extends ColumnsSelection,
+	TSelectMode extends SelectMode,
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	TPreparedQueryHKT extends PreparedQueryHKTBase,
+	TNullabilityMap extends Record<string, JoinNullability> = TTableName extends string ? Record<TTableName, 'not-null'>
+		: {},
+> extends
+	TypedQueryBuilder<
+		BuildSubquerySelection<TSelection, TNullabilityMap>,
+		SelectResult<TSelection, TSelectMode, TNullabilityMap>[]
+	>,
+	QueryPromise<SelectResult<TSelection, TSelectMode, TNullabilityMap>[]>
+{}
+
+export abstract class MySqlSetOperatorBuilder<
+	TTableName extends string | undefined,
+	TSelection extends ColumnsSelection,
+	TSelectMode extends SelectMode,
+	TPreparedQueryHKT extends PreparedQueryHKTBase,
+	TNullabilityMap extends Record<string, JoinNullability> = TTableName extends string ? Record<TTableName, 'not-null'>
+		: {},
+> extends TypedQueryBuilder<
+	BuildSubquerySelection<TSelection, TNullabilityMap>,
+	SelectResult<TSelection, TSelectMode, TNullabilityMap>[]
+> {
+	static readonly [entityKind]: string = 'MySqlSetOperatorBuilder';
+
+	protected abstract joinsNotNullableMap: Record<string, boolean>;
+	protected abstract config: {
+		fields: Record<string, unknown>;
+		limit?: number | Placeholder;
+		orderBy?: (MySqlColumn | SQL | SQL.Aliased)[];
+		offset?: number | Placeholder;
+	};
+	/* @internal */
+	abstract readonly session: MySqlSession | undefined;
+	protected abstract dialect: MySqlDialect;
+
+	/** @internal */
+	getSetOperatorConfig() {
+		return {
+			session: this.session,
+			dialect: this.dialect,
+			joinsNotNullableMap: this.joinsNotNullableMap,
+			fields: this.config.fields,
+		};
+	}
+
+	private setOperator(
```

Comment:
For consistency with select builder's `createJoin`
```suggestion
	private createSetOperator(
```